block/src/Pos/DB/Block.hs

```diff
@@ -97,15 +97,40 @@ getTipBlock = getTipSomething "block" getBlock
 getSerializedBlock
     :: forall ctx m. (MonadRealDB ctx m)
     => HeaderHash -> m (Maybe ByteString)
-getSerializedBlock hh = do
-    DB{..} <- getBlockDataDB
-    liftIO $ Rocks.get rocksDB rocksReadOpts (blockKeyMod . sformat hashHexF $ hh)
+getSerializedBlock = getSerializedBlockOrUndo fst
 
 -- Get serialization of an undo data for block with given hash from Block DB.
 getSerializedUndo :: (MonadRealDB ctx m) => HeaderHash -> m (Maybe ByteString)
-getSerializedUndo hh = do
-    DB{..} <- getBlockDataDB
-    liftIO $ Rocks.get rocksDB rocksReadOpts (undoKeyMod . sformat hashHexF $ hh)
+getSerializedUndo = getSerializedBlockOrUndo snd
+
+getSerializedBlockOrUndo ::
+       MonadRealDB ctx m
+    => ((ByteString, ByteString) -> ByteString)
+    -> HeaderHash
+    -> m (Maybe ByteString)
+getSerializedBlockOrUndo blockOrUndoSelector hh = do
+    DB {..} <- getBlockDataDB
+    let rocksGet = Rocks.get rocksDB rocksReadOpts
+    let rocksPut = Rocks.put rocksDB rocksWriteOpts
+    let rocksDel = Rocks.delete rocksDB rocksWriteOpts
+    liftIO (rocksGet (blundKeyMod hh)) >>= \case
+        Nothing -> fmap blockOrUndoSelector <$>
+            consolidateBlund rocksGet rocksPut rocksDel
+        Just ser ->
+            eitherToThrow $
+            bimap DBMalformed (Just . blockOrUndoSelector) $
```

Comment:
And now that I think of it, can't this function just return the tuple as is, and then just `fmap` it in `getSerializedUndo` and `getSerializedBlock` instead of passing selector around?

Reply:
Apparently it can, will try.